### ISSUE TITLE
Merging branch predictions

### ISSUE THREAD

**Issue by Amakri1020** [ndrplz/dreyeve#8]
Hello,

I am using this model for a similar experiment in predicting eye movement fixation maps, and was wondering what the reasoning is for simply summing and normalizing predictions from separate branches instead of doing some sort of weighted sum or concatenation?  I am still in the early stages of training this network but my image branch seems to be significantly outperforming my optical flow branch in terms of validation and test set loss - if my final multi branch architecture simply averages the outputs of the two, will it not just perform worse than if only the better performing branch is used?

**Comment by DavideA** [comment]
Hi @Amakri1020,

If I remember correctly, we took inspiration from some papers being published as we were working on it.

It is true that, as you say, the image branch will take the lead role in the prediction. That is why we envisioned a fine-tuning phase after the training of the three branches separately. In this phase, the supervision will learn how to weight each contribution. 
Without this fine-tuning, branches may interfere as you suggest.

By the way, that was just our design. I don't see an apparent reason why other strategies (e.g., concatenation, as you suggest) won't work. You can give them a try, and let us know how that goes!

D
